File: data/translated_projects/commons-fileupload/src/main/org/apache/commons/fileupload/DefaultFileItemFactory.py
from __future__ import annotations
import pathlib
import io
import os
from src.main.org.apache.commons.fileupload.disk.DiskFileItemFactory import *
import java # type: ignore
from src.main.org.apache.commons.fileupload.java_handler import *
class DefaultFileItemFactory(DiskFileItemFactory, metaclass=StaticFieldRedirector):
    def __init__(self, sizeThreshold: int, repository: pathlib.Path) -> None:

        idMapPyToJ = JavaHandler.valueToObject(dict(), "Map")
        translatedArg0 = JavaHandler.valueToObject(sizeThreshold, "int", idMapPyToJ)
        translatedArg1 = JavaHandler.valueToObject(repository, "File", idMapPyToJ)
        idMapJToPy = dict()
        try:
            self.javaObj = DefaultFileItemFactory.javaClz(translatedArg0, translatedArg1)
            self.javaObj.setPythonObject(self)
            self.javaObj.jToPy(idMapJToPy)
        except:
            raise JavaHandler.mapping(java.type("org.apache.commons.fileupload.ExceptionHandler").ERR)
        finally:
            pass
            JavaHandler.mapping(translatedArg0, idMapJToPy, sizeThreshold)
            JavaHandler.mapping(translatedArg1, idMapJToPy, repository)
    @staticmethod

    def DefaultFileItemFactory1() -> DefaultFileItemFactory:

        idMapPyToJ = JavaHandler.valueToObject(dict(), "Map")
        
        idMapJToPy = dict()
        try:
            val = JavaHandler.mapping(DefaultFileItemFactory.javaClz.DefaultFileItemFactory1(), id_map=idMapJToPy)
            return val
        except:
            raise JavaHandler.mapping(java.type("org.apache.commons.fileupload.ExceptionHandler").ERR)
        finally:
            pass

            
    javaClz = java.type("org.apache.commons.fileupload.DefaultFileItemFactory")
    def __getattr__(self, name):
        try:
            return object.__getattribute__(self, name)
        except AttributeError:
            pass
        return JavaHandler.mapping(getattr(self.javaClz, StaticFieldRedirector.unmangle_name(name)))

File: logs/glue/commons-fileupload/src/main/java/org/apache/commons/fileupload/DefaultFileItemFactory.java
package org.apache.commons.fileupload;

import java.io.File;
import org.apache.commons.fileupload.ContextInitializer;
import org.apache.commons.fileupload.ExceptionHandler;
import org.apache.commons.fileupload.IntegrationUtils;
import org.apache.commons.fileupload.disk.DiskFileItemFactory;
import org.graalvm.polyglot.PolyglotException;
import org.graalvm.polyglot.Value;

public class DefaultFileItemFactory extends DiskFileItemFactory {
  private transient Value obj =
      IntegrationUtils.createDefaultPythonObject(
          ContextInitializer.getPythonClass(
              "main/org/apache/commons/fileupload/DefaultFileItemFactory.py",
              "DefaultFileItemFactory"));

  public Value getPythonObject() {
    return obj;
  }

  public void setPythonObject(Value obj) {
    this.obj = obj;
  }

  public DefaultFileItemFactory(int sizeThreshold, File repository) {

    super(sizeThreshold, repository);
  }

  public static DefaultFileItemFactory DefaultFileItemFactory1() {

    return new DefaultFileItemFactory(0, null);
  }

  public java.util.Map pyToJ() {
    java.util.Map idMap = new java.util.HashMap();
    return pyToJ(idMap);
  }

  public java.util.Map pyToJ(java.util.Map idMap) {
    super.pyToJ(idMap);

    return idMap;
  }

  public Value jToPy() {
    Value idMap = IntegrationUtils.mapToPython(new java.util.HashMap());
    return jToPy(idMap);
  }

  public Value jToPy(Value idMap) {
    super.jToPy(idMap);
    this.obj.invokeMember("__setattr__", "javaObj", this);

    return idMap;
  }
}
